llvm/lib/Transforms/Scalar/LowerMatrixIntrinsics.cpp

```diff
@@ -269,6 +269,15 @@ computeShapeInfoForInst(Instruction *I,
       return OpShape->second;
   }
 
+  if (isa<SelectInst>(I)) {
+    auto OpShape = ShapeMap.find(I->getOperand(1));
+    if (OpShape != ShapeMap.end())
+      return OpShape->second;
+    OpShape = ShapeMap.find(I->getOperand(2));
+    if (OpShape != ShapeMap.end())
+      return OpShape->second;
```

Comment:
Picking the first operand with shape info should match what we do for uniform shapes, would probably be good to add a similar comment here (or unify the code below by only iterating over op1 & op2 for selects, via something like `drop_front(I->operands())` not sure how much extra code that would be)

Reply:
I think it should _almost_ match, but not quite: we need to ignore the condition operand for shape info iff it is not a vector.